nav2_safety_nodes/src/nav2_safety_node.cpp

```diff
@@ -0,0 +1,250 @@
+#include <chrono>
+#include <memory>
+#include <vector>
+#include <string>
+#include <utility>
+
+#include "nav2_safety_nodes/nav2_safety_node.hpp"
+
+using namespace std::chrono_literals;
+
+namespace nav2_safety_nodes
+{
+
+SafetyZone::SafetyZone()
+: nav2_util::LifecycleNode("SafetyZone", "", false)
+{
+  logger_ = get_logger();
+  RCLCPP_INFO(logger_, "Creating Safety Polygon");
+
+  // Vector of string for multiple LaserScan topics
+  const std::vector<std::string> scan{};
+  // pass polygon parameters at string
+  declare_parameter("safety_polygon", std::string("[]"));
+  declare_parameter("zone_action", rclcpp::ParameterValue(0.0));
+  declare_parameter("zone_priority", rclcpp::ParameterValue(1));
+  declare_parameter("zone_num_pts", rclcpp::ParameterValue(1));
+  declare_parameter("base_frame", std::string("base_link"));
+  declare_parameter("tf_tolerance", rclcpp::ParameterValue(0.01));
+  declare_parameter("scan_topics", scan);
+  declare_parameter("speed_limit_topic", rclcpp::ParameterValue("speed_limit"));
+}
+
+SafetyZone::~SafetyZone()
```

Comment:
If you do nothing here, then remove this from the cpp file and for the destructor in the hpp file use `= default` for the default destructor behavior